app/qml/form/FeatureToolbar.qml

```diff
@@ -1,109 +1,109 @@
-    signal editGeometryClicked()
-
-    id: toolbar
-
-    states: [
-        State {
-            name: "Edit"
-            PropertyChanges { target: editRow; visible: true }
-            PropertyChanges { target: readOnlyRow; visible: false }
-        }
-        ,State {
-            name: "Add"
-            PropertyChanges { target: editRow; visible: true }
-            PropertyChanges { target: readOnlyRow; visible: false }
-        }
-        ,State {
-            name: "ReadOnly"
-            PropertyChanges { target: editRow; visible: false }
-            PropertyChanges { target: readOnlyRow; visible: true }
-        }
-    ]
-
-    Rectangle {
-        anchors.fill: parent
-        color: InputStyle.clrPanelBackground
-    }
-
-    Row {
-        id: readOnlyRow
-        height: parent.height
-        width: parent.width
-        anchors.fill: parent
-
-        Item {
-            width: parent.width/parent.children.length
-            height: parent.height
-
-            MainPanelButton {
-                id: openProjectBtn
-                width: toolbar.itemSize
-                text: qsTr("Edit")
-                imageSource: InputStyle.editIcon
-
-                onActivated: {
-                    toolbar.editClicked()
-                    toolbar.state="Edit"
-                }
-            }
-        }
-    }
-
-    Row {
-        id: editRow
-        height: parent.height
-        width: parent.width
-        anchors.fill: parent
-
-        Item {
-            width: parent.width/parent.children.length
-            height: parent.height
-
-            MainPanelButton {
-                width: toolbar.itemSize
-                text: qsTr("Delete")
-                imageSource: InputStyle.removeIcon
-
-                onActivated: {
-                    toolbar.deleteClicked()
-                }
-            }
-        }
-
-        Item {
-            width: parent.width/parent.children.length
-            height: parent.height
-
-            MainPanelButton {
-
-                width: toolbar.itemSize
-                text: qsTr("Edit geometry")
-                imageSource: InputStyle.editIcon
-                enabled: isFeaturePoint
-
-                onActivated: {
-                    toolbar.editGeometryClicked()
-                }
-            }
-        }
-    }
-}
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+import QtQuick 2.14

```

Comment:
as I can remember, PeterP or Martin was complaining about increasing version which can resulted into some unexpected bug with QT.
On the other hand, I would like to have single max. possible version in all imports.

Reply:
Hmm, I did not know about that. @PeterPetrik @wonder-sk Shall we increase version everywhere then ( #1252 ) ? Versions in components now differ, some have 2.7, some 2.0 or 2.12 ..

Reply:
I don't recall any unexpected bug in qt...

- a selfish concern I typically have is not to use higher version number than what Qt I have on my system :-) on my system I have nowadays Qt 5.12, so I would be happy with 2.12. But as I am currently not doing much work on Input, feel free to ignore me and use the version according to Qt version that we use :-)
- a non-selfish concern is that we should ideally keep the imports consistent with our minimum required Qt version, to make sure that all components have consistent behavior and API. So e.g. if we have Qt 5.15 on all platforms, let's raise all imports to 2.15 ...

Reply:
we have 5.15 everywhere now